amr-wind/equation_systems/vof/SplitAdvection.cpp

```diff
@@ -40,126 +42,273 @@ void multiphase::split_advection(
-                amrex::Real uR = umac(i + 1, j, k);
-                multiphase::lagrangian_advection(
-                    i, j, k, 1, dtdx, uL, uR, volfrac, fluxL, fluxC, fluxR);
-            });
-        amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                multiphase::balance_fluxes(
-                    i, j, k, 1, volfrac, fluxL, fluxC, fluxR, pbc, domlo.x,
-                    domhi.x);
-            });
-        amrex::ParallelFor(
-            bxg1, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                amrex::Real vL = vmac(i, j, k);
-                amrex::Real vR = vmac(i, j + 1, k);
-                multiphase::lagrangian_advection(
-                    i, j, k, 2, dtdy, vL, vR, volfrac, fluxL, fluxC, fluxR);
-            });
-        amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                multiphase::balance_fluxes(
-                    i, j, k, 2, volfrac, fluxL, fluxC, fluxR, pbc, domlo.y,
-                    domhi.y);
-            });
-    } else if (isweep % 2 == 0) {
-        amrex::ParallelFor(
-            bxg1, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                amrex::Real vL = vmac(i, j, k);
-                amrex::Real vR = vmac(i, j + 1, k);
-                multiphase::lagrangian_advection(
-                    i, j, k, 2, dtdy, vL, vR, volfrac, fluxL, fluxC, fluxR);
-            });
-        amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                multiphase::balance_fluxes(
-                    i, j, k, 2, volfrac, fluxL, fluxC, fluxR, pbc, domlo.y,
-                    domhi.y);
-            });
-        amrex::ParallelFor(
-            bxg1, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                amrex::Real wL = wmac(i, j, k);
-                amrex::Real wR = wmac(i, j, k + 1);
-                multiphase::lagrangian_advection(
-                    i, j, k, 3, dtdz, wL, wR, volfrac, fluxL, fluxC, fluxR);
-            });
-        amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                multiphase::balance_fluxes(
-                    i, j, k, 3, volfrac, fluxL, fluxC, fluxR, pbc, domlo.z,
-                    domhi.z);
-            });
-        amrex::ParallelFor(
-            bxg1, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                amrex::Real uL = umac(i, j, k);
-                amrex::Real uR = umac(i + 1, j, k);
-                multiphase::lagrangian_advection(
-                    i, j, k, 1, dtdx, uL, uR, volfrac, fluxL, fluxC, fluxR);
-            });
-        amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                multiphase::balance_fluxes(
-                    i, j, k, 1, volfrac, fluxL, fluxC, fluxR, pbc, domlo.x,
-                    domhi.x);
-            });
+    if (is_lagrangian) {
```

Comment:
Please refactor these nested if conditions to reduce code duplication.

Reply:
See latest commit